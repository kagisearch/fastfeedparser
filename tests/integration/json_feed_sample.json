{
  "version": "https://jsonfeed.org/version/1.1",
  "title": "Example JSON Feed",
  "home_page_url": "https://example.org/",
  "feed_url": "https://example.org/feed.json",
  "description": "An example JSON Feed for testing",
  "icon": "https://example.org/icon.png",
  "authors": [
    {
      "name": "John Doe",
      "url": "https://example.org/john"
    }
  ],
  "language": "en-US",
  "items": [
    {
      "id": "https://example.org/item1",
      "url": "https://example.org/item1",
      "title": "First JSON Feed Item",
      "content_html": "<p>This is the first item in our JSON feed.</p>",
      "summary": "First item summary",
      "date_published": "2024-01-15T12:00:00Z",
      "date_modified": "2024-01-15T13:00:00Z",
      "image": "https://example.org/item1.jpg",
      "authors": [
        {
          "name": "Jane Smith"
        }
      ],
      "tags": ["test", "json", "feed"]
    },
    {
      "id": "https://example.org/item2",
      "url": "https://example.org/item2",
      "title": "Second JSON Feed Item",
      "content_text": "This is plain text content for the second item.",
      "date_published": "2024-01-14T10:30:00Z",
      "tags": ["example"]
    }
  ]
}
