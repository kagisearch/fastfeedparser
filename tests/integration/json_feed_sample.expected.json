{
  "entries": [
    {
      "author": "Jane Smith",
      "content": [
        {
          "type": "text/html",
          "value": "<p>This is the first item in our JSON feed.</p>"
        }
      ],
      "description": "First item summary",
      "id": "https://example.org/item1",
      "image": "https://example.org/item1.jpg",
      "link": "https://example.org/item1",
      "links": [
        {
          "href": "https://example.org/item1",
          "rel": "alternate",
          "type": "text/html"
        }
      ],
      "published": "2024-01-15T12:00:00+00:00",
      "tags": [
        {
          "label": null,
          "scheme": null,
          "term": "test"
        },
        {
          "label": null,
          "scheme": null,
          "term": "json"
        },
        {
          "label": null,
          "scheme": null,
          "term": "feed"
        }
      ],
      "title": "First JSON Feed Item",
      "updated": "2024-01-15T13:00:00+00:00"
    },
    {
      "content": [
        {
          "type": "text/plain",
          "value": "This is plain text content for the second item."
        }
      ],
      "description": "This is plain text content for the second item.",
      "id": "https://example.org/item2",
      "link": "https://example.org/item2",
      "links": [
        {
          "href": "https://example.org/item2",
          "rel": "alternate",
          "type": "text/html"
        }
      ],
      "published": "2024-01-14T10:30:00+00:00",
      "tags": [
        {
          "label": null,
          "scheme": null,
          "term": "example"
        }
      ],
      "title": "Second JSON Feed Item"
    }
  ],
  "feed": {
    "author": "John Doe",
    "icon": "https://example.org/icon.png",
    "id": "https://example.org/feed.json",
    "language": "en-US",
    "link": "https://example.org/",
    "links": [
      {
        "href": "https://example.org/",
        "rel": "alternate",
        "type": "text/html"
      },
      {
        "href": "https://example.org/feed.json",
        "rel": "self",
        "type": "application/json"
      }
    ],
    "subtitle": "An example JSON Feed for testing",
    "title": "Example JSON Feed"
  }
}